solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1645,6 +1650,34 @@ public Set<Path> getAllowPaths() {
     return allowPaths;
   }
 
+  /**
+   * Checks that the given path is not protected (i.e. can accept writes). Protected paths generally
+   * contain library directories that Solr loads jars and classes from.
+   *
+   * @param pathToAssert path to check
+   * @throws SolrException if path is contained inside a library folder
+   */
+  public void assertPathNotProtected(Path pathToAssert) throws SolrException {
```

Comment:
Please move this to SolrPaths.  CoreContainer has a ton of responsibilities; it doesn't need yet another one.